### PR TITLE
KWallet keyring should store service name in a folder name rather than in username

### PR DESCRIPTION
**Pull request by maciex**
This is a continuation of work from issue #83.

I rewrote all from scratch. Since last time KWallet backend was rewritten to use DBus, so most of the changes was not mergeable.

Please review.

### CODE REVIEW

**Comment by maciex** [comment]
Few assumptions I made:
1. Service name is detected from old username, it takes a string after last '@'
2. '@' is not allowed in tests (removed from DIFFICULT_CHARS)
3. Converted entries are removed
4. Empty 'Python' folder is removed

File: keyring/tests/test_backend.py [hunk 1]
```diff
@@ -14,7 +14,9 @@
 from .util import random_string
 from keyring import errors
 
-DIFFICULT_CHARS = string.whitespace + string.punctuation
```
Comment (jaraco) [inline keyring/tests/test_backend.py:17]: It's not reasonable to lower the bar of acceptability for all backends just to support a migration. If you want to override this value just for the kwallet backend, that might be reasonable, but even better would be to find a way to encode these characters such that they're not given special treatment w.r.t. other backends.

I apologize I haven't read the rest of the patch, so I don't have a good suggestion to address this concern.

**Comment by jaraco** [comment]
Thanks for working on this. I hope @mitya57 can also review and give his approval or recommendations.

File: keyring/tests/test_backend.py (hunk 1 above)
Comment (maciex) [inline keyring/tests/test_backend.py:17]: Sure, you're right. I just went the easiest path.

But I'm not sure what do you mean by "even better would be to find a way to encode these characters such that they're not given special treatment w.r.t. other backends".

I thought about overwriting that property only for migration test, would it be ok?

Comment (maciex) [inline keyring/tests/test_backend.py:17]: I moved DIFFICULT_CHARS into BackendBasicTests. '@' will be removed only for KWallet tests.

**Pull request merged**